census/views.py

```diff
@@ -60,97 +61,160 @@ def events_of(size=1000):
-                                      .order_by('start_datetime')
-        return HttpResponse(json.dumps(make_events_data_response(events)))
-
-    if 'isMonthly' in query_params and query_params['isMonthly'] == 'true':
-        # Fetch events for the whole month
-
-        month = int(query_params['month'])
-
-        # TODO: Ensure that timezone differences are properly accounted for
-        #       when using the `__month` filter
-        events = models.Event.objects.filter(approval_status=constants.EventApprovalStatus.APPROVED.name,
-                                             start_datetime__month=month)\
-                                     .order_by('start_datetime')
-        return HttpResponse(json.dumps(make_events_data_response(events)))
-
-    else:
-        # Fetch events for a selected date
-        day = query_params['day']
-        month = query_params['month']
-        year = query_params['year']
-        start_date = datetime.strptime(f"{year}-{month}-{day} 00:00:00", "%Y-%m-%d %H:%M:%S")
-        end_date = datetime.strptime(f"{year}-{month}-{day} 23:59:59", "%Y-%m-%d %H:%M:%S")
-
-        current_timezone = timezone(TIMEZONE)
-        events = models.Event.objects.filter(approval_status=constants.EventApprovalStatus.APPROVED.name,
-                                             start_datetime__range=(current_timezone.localize(start_date),
-                                                                    current_timezone.localize(end_date))) \
-            .order_by('start_datetime')
-        return HttpResponse(json.dumps(make_events_data_response(events)))
-
-
-def make_events_data_response(events):
-    result_set = {}
-    parsed_events = [parse_event_queryset(event) for event in events]
-    for month, event_data in groupby(parsed_events, key=lambda row: row['month']):
-        result_set[int(month)] = list(event_data)
-    return dict(events=result_set)
-
-
-def parse_event_queryset(event):
-    """
-    Given an Event object, return a dictionary containing only those attributes
-    that are required to be displayed.
-
-    :param event: Event objects
-    :return: A dictionary representation of the event
-    """
-
-    localized_start_datetime = event.start_datetime.astimezone(timezone(TIMEZONE))
-    start_date = datetime.strftime(localized_start_datetime, "%Y-%m-%d")
-    end_date = datetime.strftime(localized_start_datetime, "%Y-%m-%d")
-    month = datetime.strftime(localized_start_datetime, "%m")
-    start_time = datetime.strftime(localized_start_datetime, "%I:%M %p")
-    end_time = datetime.strftime(localized_start_datetime, "%I:%M %p")
-    return dict(id=event.id,
-                title=event.title,
-                description=event.description,
-                month=month,
-                start_date=start_date,
-                end_date=end_date,
-                start_time=start_time,
-                end_time=end_time,
-                )
+class HomepageView(View):
```

Comment:
I refactored this view a little bit so that it is more consistent to the way the other views are coded.